rwengine/src/audio/SoundManager.cpp

```diff
@@ -166,10 +185,81 @@ void SoundManager::SoundSource::loadFromFile(const std::string& filename) {
+
+            while ((receiveFrame = avcodec_receive_frame(codecContext, frame)) == 0) {
+                if(!swr) {
+                    if(frame->channels == 1 || frame->channel_layout == 0) {
+                        frame->channel_layout = av_get_default_channel_layout(1);
+                    }
+                    swr = swr_alloc_set_opts(nullptr,
+                                             AV_CH_LAYOUT_STEREO,                           // output channel layout
+                                             kOutputFMT,                                    // output format
+                                             frame->sample_rate,                            // output sample rate
+                                             frame->channel_layout,                         // input channel layout
+                                             static_cast<AVSampleFormat>(frame->format),    // input format
+                                             frame->sample_rate,                            // input sample rate
+                                             0,
+                                             nullptr);
+                    if (!swr) {
+                        RW_ERROR("Resampler has not been successfully allocated.");
+                        return;
+                    }
+                    swr_init(swr);
+                    if (!swr_is_initialized(swr)) {
+                        RW_ERROR("Resampler has not been properly initialized.");
+                        return;
+                    }
+                }
+
+                // Decode audio packet
+                if (receiveFrame == 0 && sendPacket == 0) {
+                    // Write samples to audio buffer
+                    resampled = av_frame_alloc();
+                    resampled->channel_layout = AV_CH_LAYOUT_STEREO;
+                    resampled->sample_rate = frame->sample_rate;
+                    resampled->format = kOutputFMT;
+                    resampled->channels = kNumOutputChannels;
+
+                    swr_config_frame(swr, resampled, frame);
+
+                    if (swr_convert_frame(swr, resampled, frame) < 0) {
+                        RW_ERROR("Error resampling "<< filename << '\n');
+                    }
+
+                    for(size_t i = 0; i < static_cast<size_t>(resampled->nb_samples) * channels; i++) {
+                        data.push_back(reinterpret_cast<int16_t *>(resampled->data[0])[i]);
+                    }
+                    av_frame_unref(resampled);
+                }
+            }
+        }
+    }
+
+#endif
 
     // Cleanup
+    /// Free all data used by the frame.
     av_frame_free(&frame);
+
+    /// Free resampler
+    swr_free(&swr);
+
+    /// Close the context and free all data associated to it, but not the context itself.
     avcodec_close(codecContext);
+
+    /// Free the context itself.
+    avcodec_free_context(&codecContext);
+
+    /// We are done here. Close the input.
```

Comment:
These comments are just repeating what the code says

Reply:
To remove?